CsharpExtras/Extensions/ArrayExtension2D.cs

```diff
@@ -1,11 +1,33 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography.X509Certificates;
 
 namespace CsharpExtras.Extensions
 {
     public static class ArrayExtension2D
     {
+        public static int LastUsedRow(this string[,] array)
+        {
+            return array.LastUsedRow(s => !string.IsNullOrWhiteSpace(s));
+        }
+        public static int LastUsedColumn(this string[,] array)
+        {
+            return array.LastUsedColumn(s => !string.IsNullOrWhiteSpace(s));
+        }
+        public static int LastUsedColumn<TVal>(this TVal[,] array, Predicate<TVal> isUsed)
+        {
+            TVal[] folded = array.FoldToSingleRow((v1, v2) => isUsed(v1) ? v1 : v2);

```

Comment:
Would it not be more efficient (and maybe clearer) to simply iterate over the entire 2D array backwards?

The way you have it implemented now you are always iterating over the entire 2D array for the fold (plus running the `isUsed` function), even if the last column is used.